modules/renter/download.go

```diff
@@ -1,805 +1,505 @@
-		// download a piece, and can be utilized again later but are currently
-		// unavailable.
-		//
-		// incompleteChunks is a list of chunks (by index) which have had a
-		// download fail. Repeat entries means that multiple downloads failed.
-		// A new worker should be assigned to the chunk for each failure,
-		// unless no more workers exist who can download pieces for that chunk,
-		// in which case the download has failed.
-		//
-		// resultChan is the channel that is used to receive completed worker
-		// downloads.
-		activePieces     int
-		activeWorkers    map[types.FileContractID]struct{}
-		availableWorkers []*worker
-		incompleteChunks []*chunkDownload
-		resultChan       chan finishedDownload
+		// Timestamp information.
+		endTime         time.Time // Set immediately before closing 'completeChan'.
+		staticStartTime time.Time // Set immediately when the download object is created.
+
+		// Basic information about the file.
+		destination       downloadDestination
+		destinationString string // The string reported to the user to indicate the download's destination.
+		destinationType   string // "memory buffer", "http stream", "file", etc.
+		staticLength      uint64 // Length to download starting from the offset.
+		staticOffset      uint64 // Offset within the file to start the download.
+		staticSiaPath     string // The path of the siafile at the time the download started.
+
+		// Retrieval settings for the file.
+		staticLatencyTarget uint64 // In milliseconds. Lower latency results in lower total system throughput.
+		staticOverdrive     int    // How many extra pieces to download to prevent slow hosts from being a bottleneck.
+		staticPriority      uint64 // Downloads with higher priority will complete first.
+
+		// Utilities.
+		log           *persist.Logger // Same log as the renter.
+		memoryManager *memoryManager  // Same memoryManager used across the renter.
+		mu            sync.Mutex      // Unique to the download object.
+	}
+
+	// downloadParams is the set of parameters to use when downloading a file.
+	downloadParams struct {
+		destination       downloadDestination // The place to write the downloaded data.
+		destinationType   string              // "file", "buffer", "http stream", etc.
+		destinationString string              // The string to report to the user for the destination.
+		file              *file               // The file to download.
+
+		latencyTarget uint64 // Workers above this latency will be automatically put on standby initially.
+		length        uint64 // Length of download. Cannot be 0.
+		needsMemory   bool   // Whether new memory needs to be allocated to perform the download.
+		offset        uint64 // Offset within the file to start the download. Must be less than the total filesize.
+		overdrive     int    // How many extra pieces to download to prevent slow hosts from being a bottleneck.
+		priority      uint64 // Files with a higher priority will be downloaded first.
```

Comment:
discreet. they'll be whatever the user sets them to, has to be the same for the whole download. Downloads for repair have priority 0. We may want to change that so that the user can choose to make their personal downloads less important than repairing, not sure.